backend/src/plugins/addons/server/services/seed/content-type-repository.ts

```diff
@@ -0,0 +1,295 @@
+         if (isMedia(value)) {
+            mediaItems.push(value.data);
+         } else if (isComponent(value)) {
+            mediaItems = mediaItems.concat(
+               this.extractMediaItemsFromComponent(value)
+            );
+         } else if (isDynamicZone(value)) {
+            mediaItems = mediaItems.concat(
+               this.extractMediaItemsFromDynamicZone(value)
+            );
+         }
+      });
+      return mediaItems;
+   }
+
+   private extractMediaItemsFromDynamicZone(
+      dynamicZone: Record<string, any>[]
+   ) {
+      let mediaItems: MediaItem[] = [];
+      for (const component of dynamicZone) {
+         mediaItems = mediaItems.concat(
+            this.extractMediaItemsFromComponent(component)
+         );
+      }
+      return mediaItems;
+   }
+
+   private extractMediaItemsFromComponent(component: Record<string, any>) {
+      let mediaItems: MediaItem[] = [];
+      forEach(component, (value, key) => {
+         if (isMedia(value)) {
+            mediaItems.push(value.data);
+         } else if (isComponent(value)) {
+            mediaItems = mediaItems.concat(
+               this.extractMediaItemsFromComponent(value)
+            );
+         } else if (isDynamicZone(value)) {
+            mediaItems = mediaItems.concat(
+               this.extractMediaItemsFromDynamicZone(value)
+            );
+         }
+      });
+      return mediaItems;
+   }
+
+   private get populateInput(): PopulateInput {
+      const attributesToBePopulated = pickBy(
+         this.schema.attributes,
+         (value) => {
+            if (value.type === 'relation') {
+               // We only need one side of the relation to be populated.
+               return value.inversedBy != null;
+            }
+            return (
+               value.type === 'dynamiczone' ||
+               value.type === 'component' ||
+               value.type === 'media'
+            );
+         }
+      );
+      const input = mapValues(attributesToBePopulated, (value) => {
+         switch (value.type) {
+            case 'relation': {
+               return {
+                  fields: '*',
+               };
+            }
+            case 'dynamiczone': {
+               return {
+                  populate: '*',
+               };
+            }
+            case 'component': {
+               return {
+                  populate: '*',
+               };
+            }
+            case 'media': {
+               return {
+                  fields: '*',
+               };
+            }
+         }
+      });
+      return input;
+   }
+}
+
+function omitNestedAttributes(attributes: Record<string, any>) {
+   return omitBy(attributes, (value) => {
+      return (
+         isRelation(value) ||
+         isComponent(value) ||
+         isDynamicZone(value) ||
+         isMedia(value)
+      );
+   });
+}
+
+function isDynamicZone(value: any) {
+   return Array.isArray(value);
+}
```

Comment:
Above, I see we type `dynamicZone` as `Record<string, any>[]`. Should this verify that each element is a component, or is that unnecessary?

```suggestion
function isDynamicZone(value: any) {
   return Array.isArray(value) && value.every((element) => isComponent(element));
}
```

Reply:
We potentially could assert that, but I'd have to test the implications. I think this plugin could use several improvements, especially in types (Strapi plugin types are not that great). 
I'd like to merge this as is, so that we can ship this and evaluate improvements in isolation. What do you think?